lxd/device/disk.go

```diff
@@ -714,9 +714,9 @@ func (d *disk) vmVirtfsProxyHelperPaths() string {
 }
 
 // vmVirtiofsdPaths returns the path for the socket and PID file to use with virtiofsd process.
-func (d *disk) vmVirtiofsdPaths() (string, string) {
-	sockPath := filepath.Join(d.inst.DevicesPath(), fmt.Sprintf("virtio-fs.%s.sock", d.name))
-	pidPath := filepath.Join(d.inst.DevicesPath(), fmt.Sprintf("virtio-fs.%s.pid", d.name))
+func (d *disk) vmVirtiofsdPaths() (sockPath string, pidPath string) {
```

Comment:
Just for my understanding - do named return values have some conditions when they are required or will that be required everywhere now?

Reply:
The linter is now checking the entire file if you have modified it. And that's one of the new rules. It allows improving the code over time.

@markylaing please correct me if I am wrong.

Reply:
Yes that's correct.

While I dislike unnecessary named returns, they are useful when a function returns multiple values of the same type. In this case, the only way to know which returned value is the socket path is to inspect the function itself or view other usages. The linter has a rule for that called `confusing-results` which is complaining here.

Reply:
> In this case, the only way to know which returned value is the socket path is to inspect the function itself or view other usages.

or a decent comment for the function ;)

But yes they do help the reader understand what is being returned.

Reply:
Ahh, so it applies only when there are multiple return values. 
Thank you both for the great explanation 👍